stix2/sources/filesystem.py

```diff
@@ -279,12 +279,12 @@ def query(self, query=None, _composite_filters=None):
                             # since ID is specified in one of filters, can evaluate against filename first without loading
                             stix_obj = json.load(open(os.path.join(root, file_)))["objects"][0]
                             # check against other filters, add if match
-                            matches = [stix_obj_ for stix_obj_ in apply_common_filters([stix_obj], query)]
+                            matches = list(apply_common_filters([stix_obj], query))
                             all_data.extend(matches)
                     else:
```

Comment:
`apply_common_filters()` returns an iterable (a generator); `list.extend()` supports an iterable parameter.  Why not pass it in directly instead of making an intermediate list?  E.g. `all_data.extend(apply_common_filters([stix_obj], query))`?  This is related to my [other comment](https://github.com/oasis-open/cti-python-stix2/issues/59#issuecomment-333225791) about trying not to make copies unnecessarily.

Reply:
No objection - I was not aware (or didnt remember) that extend() will take an iterable.